htdocs/lostpass.php

```diff
@@ -82,7 +82,7 @@
 } else {
 	$icmspass = new icms_core_Password();
 
-	$areyou = substr($getuser[0]->getVar('pass'), 0, 5);
+	$areyou = substr($getuser[0]->getVar('pass'), -5) . $getuser[0]->getVar('last_login');
```

Comment:
I think it would be more secure if `$areyou` would be hashed value

F.e.:
```php
$areyou = hash('crc32', substr($getuser[0]->getVar('pass'), -5) . $getuser[0]->getVar('last_login'));
```

This will output 8 char length code like '9064a8d0' and it will hide all components needed for that code to be generated. In current way is possible to guess that after trying many code reminders.

ChatGPT for the problem solution recommends to use md5 hash:
F.e.:
```php
$areyou = md5(substr($getuser[0]->getVar('pass'), -5) . $getuser[0]->getVar('last_login'));
```

So, code will look like '65a8e27d8879283831b664bd8b7f0ad4' and will be always 32 chars length.

First hash variant I think is easier for mobile users - if they are needed to enter code manually. The second is more secure.

Reply:
If we hash, md5 is outdated and we should use sha1 now at the least. But I do agree hashing is the way to go.

Reply:
I have added hashing to the key. I have also added a check so if the key doesn't match, it doesn't generate another email, keeping the user from being flooded if there is a brute force attack. The language constants - I've left them. They could be improved and the sprintf() can be removed. _US_RESETPWDREQ is defined and never used.

Reply:
For including icmsConfig and ICMS_URL in constant value I see only one possible problem - I think we a service that we use for easy translations - crowdin - doesn't support including variables directly into values. Idk how others do. Also that isn't compatible with i18n where we must one day to go (I think sprintf translations kinda compatible with thatmutch more)